### PR TITLE
Equipment page

### LINKED ISSUE #9458: Updates to the Equipment Page
We need to implement some changes to the Equipment page based on feedback from users, as well as to reinstate the Unequip All buttons!
screen shot 2017-11-09 at 7 15 52 pm
screen shot 2017-11-09 at 7 16 48 pm

Whoever works on this ticket will need to be added to our Zeplin so they can get all the full designs and talk with @Tressley about the details, so we'll need to know which email address to invite.

### PR DESCRIPTION
**Pull request by librarianmage**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #9458 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

I think I've worked on all that I can for the moment without some assistance.

Current issues:
* How to unequip background/pet/mount
* How to add switcher in drawer header

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID:

### CODE REVIEW

File: .nodemonignore [hunk 1]
```diff
@@ -16,4 +16,3 @@ CHANGELOG.md
 newrelic_agent.log
 *.swp
 *.swx
-website/raw_sprites/**
```
Comment (paglias) [inline .nodemonignore]: why this change? raw sprites aren't used directly in the site so what's the reason to have nodemon react to its changes?

**Comment by librarianmage** [comment]
@paglias That was a line I added myself :P @Alys suggested that I add it because it kept running out of memory/space when running the server.

**Comment by paglias** [comment]
Ah okay! If it's useful for you, feel free to keep it, it won't cause problems

File: website/client/components/inventory/equipment/index.vue [hunk 2]
```diff
@@ -34,6 +34,43 @@
...
+          :variant="costume ? 'secondary' : 'primary'",
+          class="gear-btn"
+        ) {{ $t('battleGear') }}
+        b-btn(
+          @click="costume = true",
+          :variant="costume ? 'primary' : 'secondary'",
+          class="gear-btn"
+        ) {{ $t('costume') }}
+      .float-right.clearfix
+        toggle-switch(
```
Comment (negue) [inline website/client/components/inventory/equipment/index.vue:50]: the toggle is above the unequip-buttons and not at the left side

File: website/client/components/inventory/equipment/index.vue [hunk 3]
```diff
@@ -234,6 +270,18 @@ export default {
     equipItem (item) {
       this.$store.dispatch('common:equip', {key: item.key, type: this.costume ? 'costume' : 'equipped'});
       this.gearToEquip = null;
+    },
+    unequipGear () {
+      console.log(this.activeItems);
+      for (let key in this.activeItems) {
+        console.log(this.activeItems[key]);
+        if (!this.activeItems[key].endsWith('base_0')) {
+          this.$store.dispatch('common:equip', {key: this.activeItems[key], type: this.costume ? 'costume' : 'equipped'});
```
Comment (negue) [inline website/client/components/inventory/equipment/index.vue:279]: sadly this doesn't work if you have more than one gear equipped:

the website quickly fires multiple API-calls to the server, those nearly reaches the server simultaneously, so what happens next:

```
1st request comes , gets the current gear, remove.....
   2nd request comes in, gets the current gear, removes... 
      3rd request comes in, gets the current gear, removes it, saves to db
   removes it and saves to db
removes it and saves to db
```

so at the end, only one gear was unequipped the rest is still equipped, only the website thinks "yup all unequipped" 

to fix it this, probably an own api-call would be needed to unequip all gear / costume

File: website/client/components/inventory/equipment/index.vue [hunk 4]
```diff
@@ -234,6 +270,18 @@ export default {
...
+        }
+      }
+    },
+    unequipPetMountBackground () {
+
```
Comment (negue) [inline website/client/components/inventory/equipment/index.vue:284]: I think here would be the better approach to also have a dedicated api route

File: website/client/components/ui/toggleSwitch.vue [hunk 5]
```diff
@@ -1,6 +1,11 @@
 <template lang="pug">
 .clearfix.toggle-switch-container
   .float-left.toggle-switch-description {{ label }}
+  .float-left.svg-icon.inline.icon-16.info-icon(
+    v-if="infoId",
+    v-html="icons['info']",
+    :id="infoId",
```
Comment (negue) [inline website/client/components/ui/toggleSwitch.vue:7]: is an `:id` needed for the icon?

File: website/client/components/ui/toggleSwitch.vue [hunk 6]
```diff
@@ -1,6 +1,11 @@
 <template lang="pug">
 .clearfix.toggle-switch-container
   .float-left.toggle-switch-description {{ label }}
+  .float-left.svg-icon.inline.icon-16.info-icon(
+    v-if="infoId",
+    v-html="icons['info']",
```
Comment (negue) [inline website/client/components/ui/toggleSwitch.vue:6]: here you can just use `icons.info`

**Review by negue — CHANGES_REQUESTED** [review]
I tried to compare the zeplin pages with the current implementation and tried to test the feature set itself, the first changes are on the right track, but there is a bit more to do.

What is currently missing:

- [ ] change the opacity of the costume items in the drawer, if the costume mode is deactivated
- [ ] add the different border in the alert if costume mode is deactivated
equipment_drawer
- [ ] switches in the drawer header:
   - the drawer is located in `components/ui/drawer.vue`
   - you could try to add them inside `.drawer-title`

- [ ] if you switch to the costume equipment list, but costume is active, you should be activating the mode once you select an equipment now, or else it wouldn't be shown in the current avatar / equipment list  (although the changes are sent to the server)

@Tressley should the gear stats be shown in the tooltips of `costume` gear ?

File: website/client/components/ui/toggleSwitch.vue (hunk 5 above)
Comment (librarianmage) [reply website/client/components/ui/toggleSwitch.vue:7]: That way, the pop up modal hovers on the icon

**Comment by librarianmage** [comment]
> if you switch to the costume equipment list, but costume is active, you should be activating the mode once you select an equipment now, or else it wouldn't be shown in the current avatar / equipment list (although the changes are sent to the server)

Could you elaborate on that? I'm confused

**Comment by Tressley** [comment]
@MathWhiz -- I believe @negue meant that when a user is on the Costume tab and their costume is inactive, it will set to active if they attempt to equip an item.

**Comment by negue** [comment]
@MathWhiz  sorry I didn't write an `in` before the `active` in my sentence  @Tressley cleared it up thank you ^^

**Comment by lemoness** [comment]
@MathWhiz - Hi there! How is this fix going?

**Comment by librarianmage** [comment]
I have not been able to work on it

**Comment by paglias** [comment]
any update on this one?

**Comment by librarianmage** [comment]
@paglias Hello, I completely forgot about this PR. Is it okay if I recreate the PR and finish it?

**Comment by paglias** [comment]
@MathWhiz sure!

**Comment by paglias** [comment]
Closing since @MathWhiz is going to open a new one
